mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/RowMatrix.scala

```diff
@@ -212,7 +212,7 @@ class RowMatrix(
       tol: Double,
       mode: String): SingularValueDecomposition[RowMatrix, Matrix] = {
     val n = numCols().toInt
-    require(k > 0 && k <= n, s"Request up to n singular values but got k=$k and n=$n.")
+    require(k > 0 && k <= n, s"Request up to n singular values but got k=$k and numCols=$n.")
```

Comment:
Oops, I just realized the original error message is incorrect.  It should say "Requested up to k"  (not "n").  The rest is fine.  Can you please fix this in IndexedRowMatrix as well?